src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/WaitRequestHandler.kt

```diff
@@ -187,37 +181,50 @@ class WaitRequestHandler(private val config: FrejaEidAuthenticatorPluginConfig)
                     _sessionManager.remove(SESSION_COUNTRY)
                 }
 
-                val customIdentifierAttribute = _sessionManager.get(SESSION_CUSTOM_IDENTIFIER)
-                if (customIdentifierAttribute != null)
-                {
-                    subjectAttributes.add(Attribute.of("custom-identifier", customIdentifierAttribute.value.toString()))
-                    _sessionManager.remove(SESSION_CUSTOM_IDENTIFIER)
-                }
-
                 val relyingPartyUserIdAttribute = _sessionManager.get(SESSION_RP_USER_ID)
                 if (relyingPartyUserIdAttribute != null)
                 {
                     subjectAttributes.add(Attribute.of("relyingPartyUserId", relyingPartyUserIdAttribute.value.toString()))
+                    subject = relyingPartyUserIdAttribute.value.toString()
                     _sessionManager.remove(SESSION_RP_USER_ID)
                 }
 
                 val integratorSpecificUserIdAttribute = _sessionManager.get(SESSION_INTEGRATOR_SPECIFIC_USER_ID)
                 if (integratorSpecificUserIdAttribute != null)
                 {
                     subjectAttributes.add(Attribute.of("integratorSpecificUserId", integratorSpecificUserIdAttribute.value.toString()))
+                    subject = integratorSpecificUserIdAttribute.value.toString()
                     _sessionManager.remove(SESSION_INTEGRATOR_SPECIFIC_USER_ID)
                 }
 
+                val customIdentifierAttribute = _sessionManager.get(SESSION_CUSTOM_IDENTIFIER)
```

Comment:
You could use `_sessionManager.remove()` here, since it returns the attribute when removing it.